test_data/rules/unnecessary_parenthesis.dart

```diff
@@ -131,6 +131,10 @@ main() async {
   (a?..abs())!;
   (a?[0])!;
 
+  (a?..toString())?.abs();
```

Comment:
@asashour: if you want to pick this up, that'd be awesome and it would be great to get these cases added to `UnnecessaryParenthesisTest`.

Sorry I didn't look at this PR before pushing ahead on #4042.  A lot of plates in the air...